pkg/handler/repositories.go

```diff
@@ -313,7 +313,9 @@ func (rh *RepositoryHandler) update(c echo.Context, fillDefaults bool) error {
 		return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error updating repository", err.Error())
 	}
 
-	if repoParams.URL != nil && repoConfig.URL != *repoParams.URL {
+	urlIsDifferent := repoConfig.URL != *repoParams.URL && urlUpdated
```

Comment:
What do you think moving this comparison logic into the dao Update method and using the `urlUpdated` variable to represent if the URL has changed. I believe that was the intention of that return value.

Reply:
Hmm good point. I'll do that.

Reply:
Thanks @rverdile, much cleaner.